generator/src/main/java/com/scottlogic/deg/generator/fieldspecs/FieldSpec.java

```diff
@@ -69,61 +69,56 @@ public boolean isNullable() {
     }
 
     public NumericRestrictions getNumericRestrictions() {
-        return restrictions.get(NumericRestrictions.class).orElse(null);
+        if (types == NUMERIC) {
+            return (NumericRestrictions) restrictions;
+        }
+        return null;
     }
 
     public StringRestrictions getStringRestrictions() {
-        return restrictions.get(StringRestrictions.class).orElse(null);
+        if (types == STRING) {
+            return (StringRestrictions) restrictions;
```

Comment:
What are your thoughts on making this method generic over restrictions, and passing the responsibility of the casting/conversion to the `Types` class? This would allow us to use one method, instead of three.

Reply:
i agree on doing that, but want to do it as part of a later update that makes a lot of things more generic, there are too many open branches that work on the area of restrictions currently